tests/conftest.py

```diff
@@ -0,0 +1,350 @@
+        markers = [marker.name for marker in item.iter_markers()]
+
+        if config.getoption("--4C") and "fourc" in markers:
+            selected_tests.append(item)
+
+        if config.getoption("--ArborX") and "arborx" in markers:
+            selected_tests.append(item)
+
+        if config.getoption("--CubitPy") and "cubitpy" in markers:
+            selected_tests.append(item)
+
+        if config.getoption("--Performance") and "performance" in markers:
+            selected_tests.append(item)
+
+        if not markers and not config.getoption("--exclude-standard-tests"):
+            selected_tests.append(item)
+
+    deselected_tests = list(set(items) - set(selected_tests))
+
+    items[:] = selected_tests
+    config.hook.pytest_deselected(items=deselected_tests)
+
+
+@pytest.fixture(scope="session")
+def reference_file_directory() -> Path:
+    """Provide the path to the reference file directory.
+
+    Returns:
+        Path: A Path object representing the full path to the reference file directory.
+    """
+
+    testing_path = os.path.abspath(os.path.dirname(__file__))
+    return Path(testing_path) / "reference-files"
+
+
+@pytest.fixture(scope="function")
+def current_test_name(request: pytest.FixtureRequest) -> str:
+    """Return the name of the current pytest test.
+
+    Args:
+        request: The pytest request object.
+
+    Returns:
+        str: The name of the current pytest test.
+    """
+
+    return request.node.name
+
+
+@pytest.fixture(scope="function")
+def get_string() -> Callable:
+    """Return function to get string from file.
+
+    Necessary to enable the function call through pytest fixtures.
+    """
+
+    def _get_string(path: Path) -> str:
+        """Get string from file.
+
+        Args:
+            path: Path to file.
+
+        Returns:
+            String of file
+        """
+
+        if not path.is_file():
+            raise FileNotFoundError(f"File {path} does not exist!")
+
+        with open(path, "r") as file:
+            return file.read()
+
+    return _get_string
+
+
+@pytest.fixture(scope="function")
+def compare_results() -> Callable:
+    """Return function to compare either string or files.
+
+    Necessary to enable the function call through pytest fixtures.
+    """
+
+    def _compare(
+        reference: Union[Path, str],
+        result: Union[Path, str],
+        rtol: float = 1e-14,
+        atol: float = 1e-14,
+        **kwargs,
+    ) -> bool:
+        """Comparison between reference and result with relative or absolute
+        tolerance.
+
+        Args:
+            reference: The reference string or path to the reference file.
+            result: The result string or path to the result file.
+            rtol: The relative tolerance.
+            atol: The absolute tolerance.
+
+        Returns:
+            bool: true if comparison is successful, false otherwise
+        """
+
+        if type(reference) is not type(result):
```

Comment:
This means we have to switch to `Path` objects to describe paths (at least in testing), correct? This is not used at all in the rest of MeshPy. Besides the type checking are there disadvantages of using a string here?

Reply:
Not really - in my opinion it would make sense to switch to path objects if we are at it. But let's discuss this in person


I'll just leave the first few search hints for "python string vs path" here.

https://medium.com/@guandika8/stop-representing-file-paths-in-python-using-strings-8312791f10f

https://treyhunner.com/2018/12/why-you-should-be-using-pathlib/

https://www.btelligent.com/en/blog/best-practice-working-with-paths-in-python-part-2-1/